src/gennylib/src/PoolManager.cpp

```diff
@@ -56,6 +57,15 @@ auto createPool(const std::string& name,
 mongocxx::pool::entry genny::v1::PoolManager::client(const std::string& name,
                                                      size_t instance,
                                                      const Node& context) {
+    mongocxx::pool::entry pool_entry = this->_client(name, instance, context);
+    pool_entry->list_databases();  // warm up the connection before returning it
+    return pool_entry;
+}
+
+
+mongocxx::pool::entry genny::v1::PoolManager::_client(const std::string& name,
```

Comment:
I decided to just wrap the existing pool creation code because the _apmCallback conditional duplicates the warm up code, but we can't remove the conditional yet because Genny is still using an old version of libmongoc that doesn't address the issue in TIG-1396.